### PR TITLE
Add org.dust3d.dust3d

### PR DESCRIPTION
**Pull request by huxingyi**
Related PR:
https://github.com/huxingyi/dust3d/pull/60

Related Issue:
https://github.com/huxingyi/dust3d/issues/58

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
bot, build org.dust3d.dust3d

File: org.dust3d.dust3d.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+    "app-id": "org.dust3d.dust3d",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.13",
```
Comment (LeandroStanger) [inline org.dust3d.dust3d.json:4]: Update runtime to 5.14

**Comment by bilelmoussaoui** [comment]
The build seems to fails in x86_64 because the app doesn't seem to ship an appdata/metainfo file, see the wiki for more info https://github.com/flathub/flathub/wiki/AppData-Guidelines

File: org.dust3d.dust3d.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
+{
+    "app-id": "org.dust3d.dust3d",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.14",
+    "sdk": "org.kde.Sdk",
+    "branch": "stable",
```
Comment (bilelmoussaoui) [inline org.dust3d.dust3d.json:6]:
```suggestion
```
Flathub uses stable branch by default

File: org.dust3d.dust3d.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                }
+            ]
+        },
+        {
+            "name": "dust3d",
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir -p thirdparty/instant-meshes/build",
+                "cd thirdparty/instant-meshes/build && cmake ../",
+                "cd thirdparty/instant-meshes/build && make -j 4",
```
Comment (bilelmoussaoui) [inline org.dust3d.dust3d.json:60]:
```suggestion
                "cd thirdparty/instant-meshes/build && make -j $FLATPAK_BUILDER_N_JOBS",
```

**Comment by huxingyi** [comment]
@jeteokeeffe Could you help to look into the build issues please?

**Comment by jeteokeeffe** [comment]
I'll address those issues, no worries.

**Comment by barthalion** [comment]
You can use https://bilelmoussaoui.github.io/metainfo-generator/#/ or https://www.freedesktop.org/software/appstream/metainfocreator/#/ to generate said file. You can validate it locally as described in URL Bilel has posted.

**Comment by huxingyi** [comment]
Hi, @jeteokeeffe have made some changes according to your suggestion, but the change including the appdata.xml file is not in the already released tag: "1.0.0-rc.6", could we use a hash as the sources/tag name in the json file please?

**Comment by barthalion** [comment]
> Hi, @jeteokeeffe have made some changes according to your suggestion, but the change including the appdata.xml file is not in the already released tag: "1.0.0-rc.6", could we use a hash as the sources/tag name in the json file please?

Yes, that's fine. You decide what you consider "supportable".

**Comment by jeteokeeffe** [comment]
@barthalion any update?

**Comment by jeteokeeffe** [comment]
@LeandroStanger  @bilelmoussaoui new commit, any issues?

**Comment by barthalion** [comment]
Sorry, missed your commit!

bot, build org.dust3d.dust3d

**Comment by barthalion** [comment]
Appdata file needs some changes:

```
builddir/files/share/appdata/org.dust3d.dust3d.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
• tag-missing           : <url> is not present
```

+ arm and aarch64 builds failed. They can be disabled if build failure does not ring a bell: https://github.com/flathub/flathub/wiki/App-Maintenance#limiting-the-set-of-architectures-to-build-on

**Comment by barthalion** [comment]
bot, build org.dust3d.dust3d

**Comment by barthalion** [comment]
Build fails on non-x86_64 arches. If you don't know how to fix them, [they can be disabled](https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64), but it would be nice to address the issue at some poiint.

**Comment by jeteokeeffe** [comment]
@barthalion new changes

**Comment by huxingyi** [comment]
Thanks @barthalion, we can try to see if it's an easy fix. 

Hi @jeteokeeffe, I don't know how to test flathub on aarch64 and arm, could you help to test build with the qt and image io disabled for CGAL please:
replease `cmake ../` with `cmake -D WITH_CGAL_Qt5=0 -D WITH_CGAL_ImageIO=0 ../`

**Comment by barthalion** [comment]
bot, build org.dust3d.dust3d

**Comment by barthalion** [comment]
bot, build org.dust3d.dust3d

**Comment by barthalion** [comment]
bot, build org.dust3d.dust3d

**Pull request closed without merging**